internal/batches/ui/json_lines.go

```diff
@@ -17,142 +21,167 @@ import (
-	logOperationStart("RESOLVING_NAMESPACE", "")
+	logOperationStart(LogEventOperationResolvingNamespace, nil)
 }
 func (ui *JSONLines) ResolvingNamespaceSuccess(namespace string) {
-	logOperationSuccess("RESOLVING_NAMESPACE", fmt.Sprintf("Namespace: %s", namespace))
+	logOperationSuccess(LogEventOperationResolvingNamespace, map[string]interface{}{"namespaceID": namespace})
 }
+
 func (ui *JSONLines) PreparingContainerImages() {
-	logOperationStart("PREPARING_DOCKER_IMAGES", "")
+	logOperationStart(LogEventOperationPreparingDockerImages, nil)
 }
-func (ui *JSONLines) PreparingContainerImagesProgress(percent float64) {
-	logOperationProgress("PREPARING_DOCKER_IMAGES", fmt.Sprintf("%d%% done", int(percent*100)))
+func (ui *JSONLines) PreparingContainerImagesProgress(done, total int) {
+	logOperationProgress(LogEventOperationPreparingDockerImages, map[string]interface{}{"done": done, "total": total})
 }
 func (ui *JSONLines) PreparingContainerImagesSuccess() {
-	logOperationSuccess("PREPARING_DOCKER_IMAGES", "")
+	logOperationSuccess(LogEventOperationPreparingDockerImages, nil)
 }
+
 func (ui *JSONLines) DeterminingWorkspaceCreatorType() {
-	logOperationStart("DETERMINING_WORKSPACE_TYPE", "")
+	logOperationStart(LogEventOperationDeterminingWorkspaceType, nil)
 }
 func (ui *JSONLines) DeterminingWorkspaceCreatorTypeSuccess(wt workspace.CreatorType) {
 	switch wt {
 	case workspace.CreatorTypeVolume:
-		logOperationSuccess("DETERMINING_WORKSPACE_TYPE", "VOLUME")
+		logOperationSuccess(LogEventOperationDeterminingWorkspaceType, map[string]interface{}{"type": "VOLUME"})
 	case workspace.CreatorTypeBind:
-		logOperationSuccess("DETERMINING_WORKSPACE_TYPE", "BIND")
+		logOperationSuccess(LogEventOperationDeterminingWorkspaceType, map[string]interface{}{"type": "BIND"})
 	}
 }
+
 func (ui *JSONLines) ResolvingRepositories() {
-	logOperationStart("RESOLVING_REPOSITORIES", "")
+	logOperationStart(LogEventOperationResolvingRepositories, nil)
 }
-
 func (ui *JSONLines) ResolvingRepositoriesDone(repos []*graphql.Repository, unsupported batches.UnsupportedRepoSet, ignored batches.IgnoredRepoSet) {
 	if unsupported != nil && len(unsupported) != 0 {
-		logOperationSuccess("RESOLVING_REPOSITORIES", fmt.Sprintf("%d unsupported repositories", len(unsupported)))
+		logOperationSuccess(LogEventOperationResolvingRepositories, map[string]interface{}{"unsupported": len(unsupported)})
 	} else if ignored != nil && len(ignored) != 0 {
-		logOperationSuccess("RESOLVING_REPOSITORIES", fmt.Sprintf("%d ignored repositories", len(ignored)))
+		logOperationSuccess(LogEventOperationResolvingRepositories, map[string]interface{}{"ignored": len(ignored)})
 	} else {
-		switch len(repos) {
-		case 0:
-			logOperationSuccess("RESOLVING_REPOSITORIES", "No repositories resolved")
-		case 1:
-			logOperationSuccess("RESOLVING_REPOSITORIES", "Resolved 1 repository")
-		default:
-			logOperationSuccess("RESOLVING_REPOSITORIES", fmt.Sprintf("Resolved %d repositories", len(repos)))
-		}
```

Comment:
This should be handled by the UI, not src cli (for SSBC).

Reply:
Agree, that was just lazy copy&paste from me.